app/workers/asset_manager_create_asset_worker.rb

```diff
@@ -27,6 +29,13 @@ def perform(temporary_location, asset_params, draft = false, attachable_model_cl
 
 private
 
+  def asset_already_exist?(assetable_id, assetable_type, asset_variant)
+    assetable = assetable_type.constantize.find(assetable_id)
+    if assetable
+      assetable.assets.where(variant: asset_variant).exists?
```

Comment:
this could also be:
`Asset.where(assetable_id:, assetable_type:, variant: asset_variant).exists?`